addon/services/apollo.js

```diff
@@ -26,7 +27,7 @@ function newDataFunc(observable, resultKey, resolve) {
   let mergedProps = { _apolloObservable: observable };
 
   return ({ data }) => {
-    let dataToSend = isNone(resultKey) ? data : data[resultKey];
+    let dataToSend = isNone(resultKey) ? data : get(data, resultKey);
```

Comment:
if you want to PR that separately (with a test) I'll happily merge it right away. No reason it needs to wait until I finalize this PR.